src/riak_kv_index_hashtree.erl

```diff
@@ -560,9 +561,11 @@ check_root_version(Root, Index, Version) when is_integer(Version) ->
             Version;
         false ->
             legacy
-    end;
-check_root_version(_Root, _Index, Version) ->
-    Version.
+    end
+%     ;
+% check_root_version(_Root, _Index, Version) ->
```

Comment:
Weird, you should get a type error here. Since `legacy` is a valid type for `version()` as output, it should also be valid as input and then the first clause won't match. So, that would be detectable with a type checker, one had hoped...

Reply:
Clarified types